catalog/core/catalog-core-transform-impl/src/main/java/org/codice/ddf/catalog/locator/impl/TransformerLocatorImpl.java

```diff
@@ -0,0 +1,272 @@
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+
+/**
+ * For the input transformers, first search for MultiInputTransformers, and then search for
+ * InputTransformers. That way the MultiInputTransformers will take precedence. The
+ * InputTransformers will be wrapped in a InputTransformerAdapter.
+ *
+ * <p>For the metacard transformers, first search for MultiMetacardTransformers, and then search for
+ * MetacardTransformers. That way the MultiMetacardTransformers will take precedence. The
+ * MetacardTransformers will be wrapped in a MetacardTransformerAdapter.
+ */
+public class TransformerLocatorImpl implements TransformerLocator {
+
+  @Override
+  public List<MultiInputTransformer> findMultiInputTransformers() {
+    return findMultiTransformers(
+        MultiInputTransformer.class, InputTransformer.class, InputTransformerAdapter::new, null);
+  }
+
+  @Override
+  public List<MultiInputTransformer> findMultiInputTransformers(MimeType mimeType) {
+    return findMultiTransformers(
+        MultiInputTransformer.class,
+        InputTransformer.class,
+        InputTransformerAdapter::new,
+        null,
+        multiInputTransformer -> filterByMimeType(multiInputTransformer, mimeType),
+        serviceProperties -> filterByMimeType(serviceProperties, mimeType));
+  }
+
+  @Override
+  public List<MultiInputTransformer> findMultiInputTransformers(String transformerId) {
+    return findMultiTransformers(
+        MultiInputTransformer.class,
+        InputTransformer.class,
+        InputTransformerAdapter::new,
+        generateOsgiFilter(transformerId));
+  }
+
+  @Override
+  public List<MultiMetacardTransformer> findMultiMetacardTransformers(String transformerId) {
+    return findMultiTransformers(
+        MultiMetacardTransformer.class,
+        MetacardTransformer.class,
+        MetacardTransformerAdapter::new,
+        generateOsgiFilter(transformerId));
+  }
+
+  @Override
+  public List<MultiMetacardTransformer> findMultiMetacardTransformers(MimeType mimeType) {
+    return findMultiTransformers(
+        MultiMetacardTransformer.class,
+        MetacardTransformer.class,
+        MetacardTransformerAdapter::new,
+        null,
+        multiMetacardTransformer -> filterByMimeType(multiMetacardTransformer, mimeType),
+        serviceProperties -> filterByMimeType(serviceProperties, mimeType));
+  }
+
+  @Override
+  public List<QueryResponseTransformer> findQueryResponseTransformers(MimeType mimeType) {
+    return findQueryResponseTransformers(
+        serviceProperties -> filterByMimeType(serviceProperties, mimeType), null);
+  }
+
+  @Override
+  public List<QueryResponseTransformer> findQueryResponseTransformers(String transformerId) {
+    return findQueryResponseTransformers(
+        serviceProperties -> true, generateOsgiFilter(transformerId));
+  }
+
+  @SuppressWarnings("deprecation")
```

Comment:
what's deprecated here?

Reply:
It's deprecation warning about using Constants.SERVICE_SHORTNAME. Even though it's deprecated, we want to continue using it for backwards compatibility.